File: 88.merge-sorted-array/88.js
/*
 * @lc app=leetcode id=88 lang=javascript
 *
 * [88] Merge Sorted Array
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
  var nums1_pointer = m - 1;
  var nums2_pointer = n - 1;
  var nums1_tail = m + n - 1;
  while(nums2_pointer >= 0){
    var n1 = nums1[nums1_pointer];
    var n2 = nums2[nums2_pointer];
    if((nums1_pointer >= 0) && (n1 >= n2)){
      nums1[nums1_tail] = n1;
      nums1_pointer--;
      nums1_tail--;
    } else {
      nums1[nums1_tail] = n2;
      nums2_pointer--;
      nums1_tail--;
    }
  };
};
// @lc code=end

File: 88.merge-sorted-array/88.py
class Solution:
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        index_m = m - 1
        index_n = n - 1
        index_max = m + n - 1
        while((index_m>=0) | (index_n>=0)):
            if(index_m < 0):
                nums1[index_max] = nums2[index_n]
                index_n = index_n - 1
                index_max = index_max - 1
                continue
            if(index_n < 0):
                break;
            if(nums1[index_m] >= nums2[index_n]):
                nums1[index_max] = nums1[index_m]
                index_m = index_m - 1
            else:
                nums1[index_max] = nums2[index_n]
                index_n = index_n - 1
            index_max = index_max - 1